// Copyright 2022 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";
// All generated protos should be opaque, with "xxx_hidden_" prepended to their field names.


package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";


enum BigqueryreservationBetaCapacityCommitmentPlanEnum {
  BigqueryreservationBetaCapacityCommitmentPlanEnumNO_VALUE_DO_NOT_USE = 0;
  BigqueryreservationBetaCapacityCommitmentPlanEnumCOMMITMENT_PLAN_UNSPECIFIED = 1;
  BigqueryreservationBetaCapacityCommitmentPlanEnumFLEX = 2;
  BigqueryreservationBetaCapacityCommitmentPlanEnumTRIAL = 3;
  BigqueryreservationBetaCapacityCommitmentPlanEnumMONTHLY = 4;
  BigqueryreservationBetaCapacityCommitmentPlanEnumANNUAL = 5;
}

enum BigqueryreservationBetaCapacityCommitmentStateEnum {
  BigqueryreservationBetaCapacityCommitmentStateEnumNO_VALUE_DO_NOT_USE = 0;
  BigqueryreservationBetaCapacityCommitmentStateEnumSTATE_UNSPECIFIED = 1;
  BigqueryreservationBetaCapacityCommitmentStateEnumPENDING = 2;
  BigqueryreservationBetaCapacityCommitmentStateEnumACTIVE = 3;
  BigqueryreservationBetaCapacityCommitmentStateEnumFAILED = 4;
}

enum BigqueryreservationBetaCapacityCommitmentRenewalPlanEnum {
  BigqueryreservationBetaCapacityCommitmentRenewalPlanEnumNO_VALUE_DO_NOT_USE = 0;
  BigqueryreservationBetaCapacityCommitmentRenewalPlanEnumCOMMITMENT_PLAN_UNSPECIFIED = 1;
  BigqueryreservationBetaCapacityCommitmentRenewalPlanEnumFLEX = 2;
  BigqueryreservationBetaCapacityCommitmentRenewalPlanEnumTRIAL = 3;
  BigqueryreservationBetaCapacityCommitmentRenewalPlanEnumMONTHLY = 4;
  BigqueryreservationBetaCapacityCommitmentRenewalPlanEnumANNUAL = 5;
}

message BigqueryreservationBetaCapacityCommitment {
  string name = 1;
  int64 slot_count = 2;
  BigqueryreservationBetaCapacityCommitmentPlanEnum plan = 3;
  BigqueryreservationBetaCapacityCommitmentStateEnum state = 4;
  string commitment_start_time = 5;
  string commitment_end_time = 6;
  BigqueryreservationBetaCapacityCommitmentFailureStatus failure_status = 7;
  BigqueryreservationBetaCapacityCommitmentRenewalPlanEnum renewal_plan = 8;
  string project = 9;
  string location = 10;
}

message BigqueryreservationBetaCapacityCommitmentFailureStatus {
  int64 code = 1;
  string message = 2;
  repeated BigqueryreservationBetaCapacityCommitmentFailureStatusDetails details = 3;
}

message BigqueryreservationBetaCapacityCommitmentFailureStatusDetails {
  string type_url = 1;
  string value = 2;
}

message ApplyBigqueryreservationBetaCapacityCommitmentRequest {
  BigqueryreservationBetaCapacityCommitment resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteBigqueryreservationBetaCapacityCommitmentRequest {
  string service_account_file = 1;
  BigqueryreservationBetaCapacityCommitment resource = 2;
}

message ListBigqueryreservationBetaCapacityCommitmentRequest {
  string service_account_file = 1;
  string Project = 2;
  string Location = 3;
}

message ListBigqueryreservationBetaCapacityCommitmentResponse {
  repeated BigqueryreservationBetaCapacityCommitment items = 1;
}

service BigqueryreservationBetaCapacityCommitmentService {
  rpc ApplyBigqueryreservationBetaCapacityCommitment(ApplyBigqueryreservationBetaCapacityCommitmentRequest) returns (BigqueryreservationBetaCapacityCommitment);
  rpc DeleteBigqueryreservationBetaCapacityCommitment(DeleteBigqueryreservationBetaCapacityCommitmentRequest) returns (google.protobuf.Empty);
  rpc ListBigqueryreservationBetaCapacityCommitment(ListBigqueryreservationBetaCapacityCommitmentRequest) returns (ListBigqueryreservationBetaCapacityCommitmentResponse);
}
